pkg/watermark/generic/jetstream/generic.go

```diff
@@ -115,9 +115,6 @@ func BuildSourcePublisherStores(ctx context.Context, vertexInstance *v1alpha1.Ve
 	if !vertexInstance.Vertex.IsASource() {
 		return nil, fmt.Errorf("not a source vertex")
 	}
-	if vertexInstance.Vertex.Spec.Watermark.Disabled {
```

Comment:
why?

Reply:
The Watermark.Idsabled condition check is now at the parent level
https://github.com/numaproj/numaflow/pull/1021/files#diff-685a4c165a187fb1d7c5f29325d1ac019894e70a42d915f69e394383c417b768R116